modules/templates/mailer.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package templates
+
+import (
+	"context"
+	"html/template"
+	"io/fs"
+	"os"
+	"strings"
+	texttmpl "text/template"
+
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/watcher"
+)
+
+// Mailer provides the templates required for sending notification mails.
+func Mailer(ctx context.Context) (*texttmpl.Template, *template.Template) {
+	for _, funcs := range NewTextFuncMap() {
+		subjectTemplates.Funcs(funcs)
+	}
+	for _, funcs := range NewFuncMap() {
+		bodyTemplates.Funcs(funcs)
+	}
+
+	refreshTemplates := func() {
+		for _, assetPath := range BuiltinAssetNames() {
+			if !strings.HasPrefix(assetPath, "mail/") {
+				continue
+			}
+
+			if !strings.HasSuffix(assetPath, ".tmpl") {
+				continue
+			}
+
+			content, err := BuiltinAsset(assetPath)
+			if err != nil {
+				log.Warn("Failed to read embedded %s template. %v", assetPath, err)
+				continue
+			}
+
+			assetName := strings.TrimPrefix(
+				strings.TrimSuffix(
+					assetPath,
+					".tmpl",
+				),
+				"mail/",
+			)
```

Comment:
```suggestion
			assetName := strings.TrimPrefix(strings.TrimSuffix(assetPath, ".tmpl"), "mail/")
```
Short enough to be on a single line.